### ISSUE TITLE
Redirects show the correct page but with incorrect url with ViewTransitions

### ISSUE THREAD

**Issue by hkfi** [withastro/astro#8422]
### Astro Info

```block
Astro                    v3.0.7
Node                     v18.17.0
System                   macOS (arm64)
Package Manager          pnpm
Output                   static
Adapter                  none
Integrations             @astrojs/tailwind
```

### Describe the Bug

I have set the following in my astro.config.mjs.
```
  redirects: {
    '/blog': '/blog/1'
  },
```

I have a .astro component in src/pages/blog/[page].astro. I do not have a src/pages/blog/index.astro.

When navigating to '/blog' with an anchor tag, the correct page is being shown, but the url is '/blog' instead of '/blog/1'.

### What's the expected result?

When navigating to '/blog' with an anchor tag, the url should be '/blog/1' instead of '/blog'.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-qcczt4

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by lilnasy** [comment]
https://github.com/withastro/astro/assets/69170106/27745493-fb7f-4814-ba1a-670359e9b947

This is working as you suggest it should. Navigating to `/blog` redirects the browser to `/blog/1`, and the url changes in the address bar as well. I did notice the url did not update within the stackblitz window, but that is unrelated to astro.

Feel free to reopen if you think there is still an issue.

**Comment by hkfi** [comment]
@lilnasy I think you misunderstood the issue. The problem is when navigating to `/blog` **with an anchor tag**. From the home page, you can go to `/blog` via the link that says "Blog" in the header.

**Comment by lilnasy** [comment]
You're right, I did misunderstand. I can see the issue now.

https://github.com/withastro/astro/assets/69170106/f9db5a00-8fd9-41fc-b2d7-3c79e984d4c0

**Comment by lilnasy** [comment]
@hkfi I noticed it only happens in dev mode, can you confirm?

**Comment by hkfi** [comment]
@lilnasy Yes, I'm noticing the same thing as well. 

However, in prod, there is a flicker when going to `/blog/1` via `/blog` because it goes to `/blog` first then redirecting to `/blog/1`. Is that intended? My intuition is that it would be possible to avoid the flicker with ViewTransitions or to make changes to the hrefs that matches the redirects configured during build or in the compiler.

**Comment by rishi-raj-jain** [comment]
@hkfi @lilnasy 

This seems to be fixed in the latest update of Astro.

**Comment by rishi-raj-jain** [comment]
https://stackblitz.com/edit/github-qcczt4-t8bulu?file=package.json

**Comment by lilnasy** [comment]
Thanks @rishi-raj-jain!

Closing as fixed in #8441.
